src/hooks.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from 'preact/hooks'
+
+const fetchBody = async (url, fetchOpts) => {
+  const state = {}
+  try {
+    state.response = await fetch(url, fetchOpts)
+    if (state.response.status !== 204) {
+      state.body = await state.response.text()
+    }
+    state.error = state.response.ok
+      ? undefined
+      : Error(state.response.statusText)
+  } catch (error) {
+    state.error = error
+  }
+  return state
+}
+
+export const useFetchJSON = (...args) => {
+  const state = useFetch(...args)
+  if (!state.body) return state
+  try {
+    state.data = JSON.parse(state.body)
+  } catch (error) {
+    state.error = error
+  }
+  return state
+}
+
+export const useFetch = (url, fetchOpts, inputs = [url]) => {
+  const [state, setState] = useState({ pending: true })
```

Comment:
our state is going to hold our `fetchBody` result and also tell us if it's pending or not to know when we can show the data